app/controllers/servers_controller.rb

```diff
@@ -146,6 +136,18 @@ def acces_close
     redirect_to servers_path, danger: 'Доступ запрещен !!!'
   end
 
+  def search_params
+    client_timezone = params[:client_timezone].to_i.hours
+    datestart_begin = params[:datestart_begin].to_datetime - client_timezone if params[:datestart_begin].present?
+    datestart_end = params[:datestart_end].to_datetime.end_of_day - client_timezone if params[:datestart_end].present?
+    search_params = {}
+    search_params.tap do |hash|
```

Comment:
we can not introduce the new variable - do just 
`{}.tap do |hash|` 
(remove search_params)